gui/scripts/syncthing/core/controllers/syncthingController.js

```diff
@@ -321,6 +321,10 @@ angular.module('syncthing.core')
         function refreshSystem() {
             $http.get(urlbase + '/system').success(function (data) {
                 $scope.myID = data.myID;
+
+                // convert amount of seconds to string format "d h m s" without zero values
+                data.uptime = ('0' + Math.floor(data.uptime/86400) + 'd ' + (new Date(data.uptime%86400*1000)).toUTCString().replace(/.*(\d\d):(\d\d):(\d\d).*/, "$1h $2m $3s")).replace(/(\s?\b00\w)|(\b0)/g, '').trim();
```

Comment:
If we're going to be this precise with the time, it probably needs to update each second? And this looks like it should be at least it's own function, maybe a filter?